src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -586,23 +586,29 @@ public CatalogObjectMetadata restore(String bucketName, String name, Long commit
         return new TreeSet<>(catalogObjectRepository.findAllContentTypes());
     }
 
-    public List<CatalogObjectNameReference> getAccessibleCatalogObjectsNameReferenceByKind(boolean sessionIdRequired,
-            String sessionId, Optional<String> kind) {
-        List<CatalogObjectNameReference> catalogObjectsNameReferenceByKind;
-        if (!kind.isPresent()) {
-            catalogObjectsNameReferenceByKind = generateCatalogObjectsNameReferenceByKind(catalogObjectRepository.findAllCatalogObjectNameReferenceByKind());
+    public List<CatalogObjectNameReference> getAccessibleCatalogObjectsNameReferenceByKindAndContentType(
+            boolean sessionIdRequired, String sessionId, Optional<String> kind, Optional<String> contentType) {
+        List<CatalogObjectNameReference> catalogObjectsNameReferenceByKindAndContentType;
+        if (kind.isPresent() && contentType.isPresent()) {
```

Comment:
All the following part can be improved/replaced by just calling one method:
 `
 catalogObjectsNameReferenceByKindAndContentType = generateCatalogObjectsNameReferenceByKind(
                catalogObjectRepository.findCatalogObjectNameReferenceByKindAndContentType(
                        kind.orElse(""),
                        contentType.orElse("")));
`
I've checked that in case of Optional empty it will pass "" String as argument. It works fine for `LIKE lower(concat('%', ?2, '%'))"`
In general try to avoid many if-else conditions. It makes the design more clear and easier to maintain.